src/basic/sysctl-util.c

```diff
@@ -193,3 +194,16 @@ int sysctl_read_ip_property(int af, const char *ifname, const char *property, ch
 
         return sysctl_read(p, ret);
 }
+
+int sysctl_read_ip_property_uint32(int af, const char *ifname, const char *property, uint32_t *ret) {
+        _cleanup_free_ char *s = NULL;
+        int r;
+
+        assert(ret);
+
+        r = sysctl_read_ip_property(af, ifname, property, &s);
+        if (r < 0)
+                return r;
+
+        return safe_atou32(s, ret);
+}
```

Comment:
maybe, add:

```c
#define sysctl_read_ip_property_uint(af,ifname,property,ret) sysctl_read_ip_property_uint32(af,ifname,property,ret)
```

and then use that in routes_max() in src/network/networkd-route-util.c.